CONTRIBUTING.md

```diff
@@ -239,7 +230,6 @@ only pull requests targeted directly against master.
   - **no PRs targeting this branch should be merged unless exceptional circumstances arise**
 - On the `RC` branch, prepare a new version section in the `CHANGELOG.md`
   - All links must be link-ified: `$ python ./scripts/linkify_changelog.py CHANGELOG.md`
-  - Copy the entries into a `RELEASE_CHANGELOG.md`, this is needed so the bot knows which entries to add to the release page on github.
```

Comment:
@alessio is this still true? If so I can add it back, but I've never seen a release changelog

Reply:
I'd say no actually, it isn't. @marbar3778 added this but I believe we never actually respected it :-/

Reply:
it still works, just no one uses it.

Reply:
I'll add it back